api/src/database/index.ts

```diff
@@ -127,3 +129,17 @@ export async function isInstalled(): Promise<boolean> {
 	// exists when using the installer CLI.
 	return await inspector.hasTable('directus_collections');
 }
+
+export const validateMigrations = async (): Promise<boolean> => {
+	const database = getDatabase();
+	try {
+		const completedMigrations = await database.select<Migration[]>('*').from('directus_migrations').orderBy('version');
+		const completedMigrationFiles = completedMigrations.map((migration) => {
+			return `${migration.version}-${migration.name.toLowerCase()}.ts`.replace(/ /g, '-');
+		});
+		const migrationFiles = await fse.readdir('./migrations');
+		return completedMigrationFiles.every((migration) => migrationFiles.includes(migration));
+	} catch (error) {
+		return error;
+	}
```

Comment:
Instead of catching and returning the error (and then not handling it anywhere else), we should probably just have it throw the process here. The only thing that we expect that might fail here is the read to `./migrations`, in which case there's a way bigger problem that's OK to crash for